app/controllers/users.py

```diff
@@ -106,6 +106,46 @@ def get(self):
             data=dict(users=json.loads(users_data))
         ), 200
 
+class UserAdimUpdateView(Resource):
+    
+    @admin_required
+    def patch(self):
+        try:
+            user_schema = UserSchema(only=("is_beta_user",))
+
+            user_data = request.get_json()
+            user_id = user_data["user_id"]
+            
+            validate_user_data, errors = user_schema.load(user_data)
+
+            if errors:
+                return dict(status='fail', message=errors), 400
+
+            
+            user = User.get_by_id(user_id)
+            if not user:
+                return dict(
+                    status='fail',
+                    message=f'User {user_id} not found'
+                ), 404
+
+            updated = User.update(user, **validate_user_data)
+
+            if not updated:
+                return dict(
+                    status='fail',
+                    message='Internal Server Error'
+                ), 500
+
+            return dict(
+                status='success',
+                message=f'User {user_id} updated successfully'
+            ), 200
+
+        except Exception as e:
```

Comment:
nice exception handling here. maybe in the near future we could try to return an exact error instead of stringifying what is thrown by the server

Reply:
This is noted.